### PR TITLE
Get the funq out of here

### PR DESCRIPTION
**Pull request by david-chenevert-zocdoc**
:warning: no merge, no review

### CODE REVIEW

File: src/ServiceStack.Plugins.ProtoBuf/packages.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="protobuf-net" version="2.0.0.640" targetFramework="net35" />
+  <package id="protobuf-net" version="2.0.0.640" targetFramework="net35" requireReinstallation="True" />
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack.Plugins.ProtoBuf/packages.config]: What's this about?

File: src/ServiceStack.ServiceInterface/ServiceExtensions.cs [hunk 2]
```diff
@@ -234,5 +237,6 @@ public static IAuthSession GetSession(this IHttpRequest httpReq, bool reload = f
             var response = runner.Execute(requestContext, service, request);
             return response;
         }
+        */
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack.ServiceInterface/ServiceExtensions.cs]: Consider deleting code instead of commenting it out.

File: src/ServiceStack.ServiceInterface/Testing/BasicAppHost.cs [hunk 3]
```diff
@@ -140,4 +141,5 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
     }
-}
\ No newline at end of file
+}
+*/
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack.ServiceInterface/Testing/BasicAppHost.cs:145]: Consider deleting code instead of commenting it out.

File: src/ServiceStack.ServiceInterface/Validation/ValidationFeature.cs [hunk 4]
```diff
@@ -52,28 +52,23 @@ public static class ValidationExtensions
     {
         /// <summary>
         /// Auto-scans the provided assemblies for a <see cref="IValidator"/>
-        /// and registers it in the provided IoC container.
+        /// and registers it in the provided IoC dependencyInjector.
         /// </summary>
-        /// <param name="container">The IoC container</param>
+        /// <param name="dependencyInjector">The IoC dependencyInjector</param>
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack.ServiceInterface/Validation/ValidationFeature.cs]: This param name doesn't match what you have in the method signature.

File: src/ServiceStack/ServiceHost/DependencyInjectorResolveCache.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+using System.Text;
+using Autofac;
+using ServiceStack.DependencyInjection;
+using ServiceStack.Configuration;
+using System.Linq.Expressions;
+using System.Threading;
+
+namespace ServiceStack.ServiceHost
+{
+    public class DependencyInjectorResolveCache : ITypeFactory
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack/ServiceHost/DependencyInjectorResolveCache.cs]: This class doesn't cache anything, correct?

File: src/ServiceStack/ServiceHost/ServiceController.cs [hunk 6]
```diff
@@ -549,4 +553,4 @@ public void AssertServiceRestrictions(Type requestType, EndpointAttributes actua
         }
     }
 
-}
\ No newline at end of file
+}
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack/ServiceHost/ServiceController.cs:527]: This file is where the magic is.

File: src/ServiceStack/ServiceHost/ServiceManager.cs [hunk 7]
```diff
@@ -77,15 +77,16 @@ private List<Type> GetAssemblyTypes(Assembly[] assembliesWithServices)
 			}
 		}
 
-		private ContainerResolveCache typeFactory;
+        private DependencyInjectorResolveCache typeFactory;
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack/ServiceHost/ServiceManager.cs]: tabs

File: src/ServiceStack/ServiceHost/ServiceManager.cs [hunk 8]
```diff
@@ -136,10 +137,6 @@ public object Execute(object dto)
 
 		public void Dispose()
 		{
-			if (this.Container != null)
-			{
-				this.Container.Dispose();
-			}
```
Comment (kyle-fritz-zocdoc) [inline src/ServiceStack/ServiceHost/ServiceManager.cs:142]: Why is this deleted?

File: src/ServiceStack/ServiceHost/DependencyInjectorResolveCache.cs (hunk 5 above)
Comment (david-chenevert-zocdoc) [inline src/ServiceStack/ServiceHost/DependencyInjectorResolveCache.cs]: I deleted it.

File: src/ServiceStack/ServiceHost/ServiceManager.cs (hunk 8 above)
Comment (david-chenevert-zocdoc) [inline src/ServiceStack/ServiceHost/ServiceManager.cs:142]: The Dispose() logic in Funq.Container and Autofac and my new class are all very different.
This Dispose() may have made sense in the Funq world, but is not relevant for DependencyService.
